windup/windup-operator
# operator-005-mta-514

Publish Operator in operatorhub for MTA 5.1.4
https://issues.redhat.com/browse/WINDUP-3046


Test case :

0. install Golang
0. clone  https://github.com/operator-framework/operator-registry ( if you haven't yet )
1. build it  ` cd operator-registry && make build`
2. it will generate `bin` folder with `opm` binary
2. clone this PR
1. create an Openshift 4.6.x environment and log to it with oc login ( find the command on ? in the web console )
1. podman pull quay.io/openshift-community-operators/catalog:latest
2. create and install catalog for version 0.0.4 , you should have this version already on your quay repository
./create-catalog.sh -u {your quayuser} -p {your quay pwd} -m 0.0.4 -b 0

3. install operator from the test catalog on namespace mta

4. create a new MTA instance

Expect:

1. when creating a new instance the MTA version on the field is defaulted to MTA 5.1.3.Final

Test Case :

1.create and install catalog for version 0.0.5, you will build current code
./create-catalog.sh -u {your quayuser} -p {your quay pwd} -m 0.0.5 -b 1

2. check that the installed operator will be upgraded to 0.0.5

3. delete the mta resource created before ( installed operators -> mta operator -> MTA instances -> :delete ) to free resources

4. create a new MTA instance

Expect:

1. when creating a new instance the MTA version on the field is defaulted to MTA 5.1.4.Final
